### PR TITLE
all: implement more of the webgpu.idl

### PR DESCRIPTION
**Pull request by gabrielschulhof**
Signed-off-by: @gabrielschulhof

### CODE REVIEW

File: test/promise/promise-impl.cc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#include <stdio.h>
+#include "promise-impl.h"
+
+Promise<FulfillsPromise>
+ReturnsPromise::requestPromise(DOMString name) {
+  FulfillsPromise result{name};
+  Promise<FulfillsPromise> promise;
+  promise.Resolve(result);
```
Comment (cerisano) [inline test/promise/promise-impl.cc:8]: Seems to be returning `undefined` from `await`

**Review by cerisano — COMMENTED** [review]
Approved promise draft (see note below).

**Review by cerisano — COMMENTED** [review]
Tested this PR patched to main branch.
Not seeing the errors in the CI (`node test.js ` in `test/promise`  passes).

There seems to be a link error when running `node test.js ` in `test/webgpu`,
but I assume this will be addressed in the draft.

```
GPU::RequestAdapter with options { powerPreference: 1(High_performance) }
node: symbol lookup error: /home/dcerisano/git/webidl-napi/test/webgpu/build/Release/webgpu.node: undefined symbol: _ZN10WebIdlNapi9ConverterI10GPUAdapterE4ToJSEP10napi_env__RKS1_PP12napi_value__
```

**Comment by gabrielschulhof** [comment]
The scope of the PR has expanded a bit:
  * Promise
  * FrozenArray
  * sequence
  * interface attributes (via property accessors)

**Comment by cerisano** [comment]
Could lead to some extension of N-API itself, which is a stated goal and also in scope.

**Comment by cerisano** [comment]
Also for reference, here are [Chromium's webidl definitions for WebGPU](https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/renderer/modules/webgpu/) (*.idl)

**Review by cerisano — APPROVED** [review]

**Pull request closed without merging**